bootstrapping/bootstrapping.go

```diff
@@ -128,20 +133,27 @@ func stop(c *cli.Context) error {
 }
 
 // Main bootstrapping background routine
-func bootstrappingRoutine(ctx context.Context, c *cli.Context, timingInterval int64) {
+func bootstrappingRoutine(ctx context.Context, c *cli.Context, timingInterval int64, thresholdLines int) {
 	log.Debug("bootstrappingRoutine")
 
-	//formatter := format.GetFormatter()
 	bootstrappingSvc, formatter := cmd.WireUpBootstrapping(c)
-
-	// initialization
+	commandProcessed := make(chan bool, 1)
+	isRunningCommandRoutine := false
 	currentTicker := time.NewTicker(time.Duration(timingInterval) * time.Second)
 	for {
-		go processingCommandRoutine(bootstrappingSvc, formatter)
+		if !isRunningCommandRoutine {
+			isRunningCommandRoutine = true
+			go processingCommandRoutine(bootstrappingSvc, formatter, thresholdLines, commandProcessed)
+		}
 
-		log.Debug("Waiting...", currentTicker)
+		log.Debug("waiting...", currentTicker)
 
 		select {
+		case <-commandProcessed:
+			isRunningCommandRoutine = false
+			currentTicker.Stop()
+			currentTicker = time.NewTicker(time.Duration(timingInterval) * time.Second)
+			log.Debug("command processed")
```

Comment:
This may work, but it is more complex than it needs to be: why do we need the command run by another goroutine?

Reply:
In the command-polling scenario it made sense because we wanted to ensure a regular-interval polling, but this is not the case here